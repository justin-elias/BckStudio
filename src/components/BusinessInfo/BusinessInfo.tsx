// Core
import React from "react";
// Style
import {makeStyles} from "@mui/styles";
//@ts-ignore
import classNames from "classnames";
import {container, title, dividerBar, mainElement} from "../../assets/globals";
//Components
import GridContainer from "../Grid/GridContainer";
import GridItem from "../Grid/GridItem";
import {Hidden, Typography} from "@mui/material";
import Image from "next/image";
// Assets

/* eslint-disable no-unused-vars */
const useStyles = makeStyles((theme) => ({
    /* eslint-enable no-unused-vars */

    container,
    title,
    dividerBar,
    mainElement,
    inlineBlock: {
        display: "inline-block",
        padding: "0px",
        width: "auto",
    },
    icon: {
        width: "18px",
        height: "18px",
        position: "relative",
        top: "3px"
    },
    centered: {
        textAlign: "center"
    },
    verticalCones: {
        height: "10rem"
    },
    inline: {
        display: "inline"
    },
    vertCones: {
        marginTop: "2rem"
    },
    horizCones: {
        marginTop: "2rem"
    },
    phone: {
        display: "inline",
    },
    phoneLink: {
        display: "inline",
    },
    center: {
        maxWidth: "12rem",
        margin: "auto"
    }
}));

export default function BusinessInfo(props: any) {
    const classes = useStyles(props);
    return (
        <div>
            <div className={classes.dividerBar}/>
            <GridContainer spacing={0} justifyContent={"center"} aria-label={'Business Info'}>
                <GridItem xs={12} sm={12} md={3} aria-label={'Business Hours'}>
                    <Typography variant={"h3"} className={classNames(classes.title, classes.centered)}>HOURS</Typography>
                    <Typography variant={"body1"} className={classes.centered}> Tuesday &amp; Wednesday 8:30AM - 5:30PM<br/>Thursday -
                        Saturday 9:00AM - 6:30PM<br/>Closed Sunday &amp; Monday</Typography>
                </GridItem>
                <Hidden smDown>
                    <div className={classes.vertCones}>
                        <Image src={"https://media.graphassets.com/hfaJUnYBRIC8AySsGbkB"} alt={"Vertical divider which shows temperature cones at various stages of drooping"} height={"160"} width={"12"} unoptimized/>
                    </div>
                </Hidden>
                <Hidden mdUp>
                    <div className={classes.horizCones}>
                        <Image src={"https://media.graphassets.com/MFyMrP7IQisembYJM4k6"} alt={"Horizontal divider which shows temperature cones at various stages of drooping"} height={"32"} width={"240"} unoptimized/>
                    </div>
                </Hidden>
                <GridItem xs={12} sm={12} md={3} aria-label={'Business Address'}>
                    <Typography variant={"h3"} className={classNames(classes.title, classes.centered)} align={"center"}>LOCATION</Typography>
                    <Typography variant={"body1"} className={classes.centered} align={"center"}>6062 Jackrabbit Lane<br/>
                        Belgrade, Montana 59714
                    </Typography>
                    <div className={classes.center}>
                        <Typography variant={"h6"} component={"p"} className={classNames(classes.title, classes.phone)} align={"center"} justifySelf={"center"}>
                            Phone:
                        </Typography>
                        <Typography variant={"body1"} className={classes.phoneLink} align={"center"}>
                            <a href={"tel:+14062249229"}> (406)224-9229</a>
                        </Typography>
                    </div>
                </GridItem>
            </GridContainer>
        </div>
    )
}