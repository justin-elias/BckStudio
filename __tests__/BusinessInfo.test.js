import React from "react";
import '@testing-library/jest-dom';
import {render, getByLabelText} from "@testing-library/react";
import BusinessInfo from '../src/components/BusinessInfo/BusinessInfo';


describe("BusinessInfo", () => {
    test('should match snapshot', () => {
        const {asFragment} = render(<BusinessInfo/>);
        expect(asFragment()).toMatchSnapshot();
    });

    test("should render the BusinessInfo component", () => {
        const {getByLabelText} = render(<BusinessInfo/>);
        const businessInfo = getByLabelText("Business Info");
        expect(businessInfo).toBeInTheDocument();
    });

    test('should render Business Hours Component', () => {
        const {getByLabelText} = render(<BusinessInfo/>);
        const businessHours = getByLabelText("Business Hours");
        expect(businessHours).toBeInTheDocument();
    });

    test("should render the BusinessInfo component with the Hours text", () => {
        const {getByText} = render(<BusinessInfo/>);
        const businessInfo = getByText("HOURS");
        expect(businessInfo).toBeInTheDocument();
    });

    test('should render Business Address component', () => {
        const {getByLabelText} = render(<BusinessInfo/>);
        const businessAddress = getByLabelText("Business Address");
        expect(businessAddress).toBeInTheDocument();
    });
    test("should render the BusinessInfo component with the Location text", () => {
        const {getByLabelText, getByText} = render(<BusinessInfo/>);
        const businessInfo = getByText("LOCATION");
        expect(businessInfo).toBeInTheDocument();
    });
});