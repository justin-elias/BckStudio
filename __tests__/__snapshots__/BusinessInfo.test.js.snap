// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessInfo should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-dividerBar-3"
    />
    <div
      aria-label="Business Info"
      class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
    >
      <div
        aria-label="Business Hours"
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3 makeStyles-grid-15 undefined css-iv7vbz-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 makeStyles-title-2 makeStyles-centered-7 css-gepadz-MuiTypography-root"
        >
          HOURS
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 makeStyles-centered-7 css-ahj2mt-MuiTypography-root"
        >
           Tuesday & Wednesday 8:30AM - 5:30PM
          <br />
          Thursday - Saturday 9:00AM - 6:30PM
          <br />
          Closed Sunday & Monday
        </p>
      </div>
      <div
        aria-label="Business Address"
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3 makeStyles-grid-15 undefined css-iv7vbz-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter makeStyles-title-2 makeStyles-centered-7 css-47suzj-MuiTypography-root"
        >
          LOCATION
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-centered-7 css-1uwgr7b-MuiTypography-root"
        >
          6062 Jackrabbit Lane
          <br />
          Belgrade, Montana 59714
        </p>
        <div
          class="makeStyles-center-14"
        >
          <p
            class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter makeStyles-title-2 makeStyles-phone-12 css-d5ls3n-MuiTypography-root"
          >
            Phone:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-phoneLink-13 css-1uwgr7b-MuiTypography-root"
          >
            <a
              href="tel:+14062249229"
            >
               (406)224-9229
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
